src/rgw/rgw_lc_s3.cc

```diff
@@ -151,10 +156,6 @@ void LCFilter_S3::decode_xml(XMLObj *obj)
     obj_tags.emplace_tag(std::move(_key), std::move(_val));
     num_conditions++;
   }
-
-  if (single_cond && num_conditions > 1) {
```

Comment:
it looks like these variables can be removed entirely

Reply:
yes, they are not needed

Reply:
will review dump_xml();  afaik, yes